pennylane/data/base/attribute.py

```diff
@@ -0,0 +1,453 @@
+        be created in an existing HDF5 group by passing the ``parent_and_key``
+        argument.
+
+        Bind initialization loads an attribute that was previously persisted
+        in HDF5 object ``bind``.
+
+        Note that if ``bind`` is provided, all other arguments will be ignored.
+
+        Args:
+            value: Value to initialize attribute to
+            info: Metadata to attach to attribute
+            bind: HDF5 object from which existing attribute will be loaded
+            parent_and_key: A 2-tuple specifying the HDF5 group that will contain
+                this attribute, and its key.
+        """
+        if bind is not None:
+            self._bind = bind
+            self._check_bind()
+            return
+
+        if parent_and_key is not None:
+            parent, key = parent_and_key
+        else:
+            parent, key = h5py.group(), "_"
+
+        if value is UNSET:
+            value = self.default_value()
+            if value is UNSET:
+                raise TypeError("__init__() missing 1 required positional argument: 'value'")
+
+        self._bind = self._set_value(value, info, parent, key)
+        self._check_bind()
+        self.__post_init__(value, self.info)
+
+    @property
+    def info(self) -> AttributeInfo:
+        """Returns the ``AttributeInfo`` for this attribute."""
+        return AttributeInfo(self.bind.attrs)
+
+    @property
+    def bind(self) -> HDF5:
+        """Returns the HDF5 object that contains this attribute's
+        data."""
+        return self._bind
+
+    @classmethod
+    def default_value(cls) -> Union[InitValueType, Literal[UNSET]]:
+        """Returns a valid default value for this type, or ``UNSET`` if this type
+        must be initialized with a value."""
+        return UNSET
+
+    @classmethod
+    def py_type(cls, value_type: Type[InitValueType]) -> str:
+        """Determines the ``py_type`` of an attribute during value initialization,
+        if it was not provided in the ``info`` argument. This method returns
+        ``f"{value_type.__module__}.{value_type.__name__}``.
+        """
+        return get_type_str(value_type)
+
+    @classmethod
+    def consumes_types(cls) -> typing.Iterable[type]:
+        """
+        Returns an iterable of types for which this should be the default
+        codec. If a value of one of these types is assigned to a Dataset
+        without specifying a `type_id`, this type will be used.
+        """
+        return ()
+
+    def __post_init__(self, value: InitValueType, info: Optional[AttributeInfo]) -> None:
+        """Called after __init__(), only during value initialization. Can be implemented
+        in subclasses that require additional initialization."""
+
+    def get_value(self) -> ValueType:
+        """Deserializes the mapped value from ``bind``."""
+        return self.hdf5_to_value(self.bind)
+
+    def copy_value(self) -> ValueType:
```

Comment:
What do you mean by "advertise `get_value()`"?

Reply:
as in, put it on our public docs. My main question is, given that this function does the same thing as `get_value()`, is it necessary to include? if it's fulfilling some API requirement, then nbd. but otherwise, we can just remove it and tell users to call `get_value` if they need to get a value (copied or not, since both copy it anyway)